vita3k/kernel/src/thread/thread.cpp

```diff
@@ -280,24 +263,30 @@ int run_callback(KernelState &kernel, ThreadState &thread, const SceUID &thid, A
     // TODO: remaining arguments should be pushed into stack
     set_thread_id(*cpu, thid);
     write_pc(*cpu, callback_address);
-    CPUStatePtr old_thread_cpu = CPUStatePtr(cpu, [](CPUState *state) -> void {});
-    old_thread_cpu.swap(thread.cpu);
-    auto res = run(*cpu, true, callback_address);
-    thread.cpu.swap(old_thread_cpu);
+    write_lr(*cpu, cpu->halt_instruction_pc);
+    auto res = run(*cpu);
+
```

Comment:
see https://github.com/Vita3K/Vita3K/commit/892f88cbff20d7439f00960d3ecd5ae67823f8ae
thread.cpu swap is important when callback is called from callback and both callbacks use stack. Check on atelier game